python/rmm/_lib/device_buffer.pyx

```diff
@@ -179,6 +183,51 @@ cdef class DeviceBuffer:
         with nogil:
             copy_host_to_ptr(hb[:s], <uintptr_t>dbp.data(), stream)
 
+    cpdef copy_from_device(self, cuda_ary, uintptr_t stream=0):
+        """Copy from a buffer on host to ``self``
+
+        Parameters
+        ----------
+        cuda_ary : object to copy from that has ``__cuda_array_interface__``
+        stream : CUDA stream to use for copying, default 0
+
+        Examples
+        --------
+        >>> import rmm
+        >>> db = rmm.DeviceBuffer(size=5)
+        >>> db2 = rmm.DeviceBuffer.to_device(b"abc")
+        >>> db.copy_from_device(db2)
+        >>> hb = db.copy_to_host()
+        >>> print(hb)
+        array([97, 98, 99,  0,  0], dtype=uint8)
+        """
+        if not hasattr(cuda_ary, "__cuda_array_interface__"):
+            raise RuntimeError(
+                "Expected object to support `__cuda_array_interface__` "
+                "protocol"
+            )
+
+        cdef device_buffer* dbp = self.c_obj.get()
+
+        cuda_ary_interface = cuda_ary.__cuda_array_interface__
+        cdef uintptr_t src_ptr = cuda_ary_interface["data"][0]
+        cdef size_t s = (
+            reduce(mul, cuda_ary_interface["shape"])
+            * np.dtype(cuda_ary_interface["typestr"]).itemsize
+        )
```

Comment:
Should we just require the user to supply us a 1-D contiguous array and error out otherwise? This would match the other functions. Also seems simpler to handle here. Plus they have lots of options to make this work like `.ravel(...)` on arrays.

Also what do you think about enforcing type as well?

Reply:
I don't think we should enforce type given not every library supports viewing as different dtypes. I'm all for enforcing it's a 1-D contiguous array though 👍

Reply:
Yeah that makes sense. Sounds good 🙂

Reply:
Pushed an update to resolve this.